### PR TITLE
make confidence interval work with length mismatch

### PR DESCRIPTION
**Pull request by FrankKr**
Fixes 
image

### CODE REVIEW

File: openstef/model/confidence_interval_applicator.py [hunk 1]
```diff
@@ -204,10 +204,14 @@ def _add_quantiles_to_forecast_quantile_regression(
...
+        quantile_df = pd.DataFrame(index=self.forecast_input_data.index)
         for quantile in quantiles:
             quantile_key = f"quantile_P{quantile * 100:02.0f}"
-            forecast[quantile_key] = self.model.predict(
+            quantile_df[quantile_key] = self.model.predict(
                 self.forecast_input_data, quantile=quantile
             )
 
-        return forecast
+        return forecast.merge(
```
Comment (JanMaartenvanDoorn) [inline openstef/model/confidence_interval_applicator.py:215]: Looks good!

File: test/unit/model/test_confidence_interval_applicator.py [hunk 2]
```diff
@@ -105,6 +106,23 @@ def test_add_quantiles_to_forecast(self):
         for expected_column in expected_new_columns:
             self.assertTrue(expected_column in pp_forecast.columns)
 
+    def test_add_quantiles_to_forecast_length_mismatch(self):
+        pj = {"quantiles": self.quantiles}
+        pp_forecast = ConfidenceIntervalApplicator(
+            MockModel(), self.stdev_forecast.iloc[:-1, :]  # do not use last value
+        )._add_quantiles_to_forecast_quantile_regression(
```
Comment (JanMaartenvanDoorn) [inline test/unit/model/test_confidence_interval_applicator.py:113]: Can you please add the arrange, act assert comments? And maybe A one line docstring about what is tested.

**Review by JanMaartenvanDoorn — APPROVED** [review]
LGTM!

File: test/unit/model/test_confidence_interval_applicator.py (hunk 2 above)
Comment (FrankKr) [reply test/unit/model/test_confidence_interval_applicator.py:113]: Good idea!

**Pull request merged**